034-default-to-restricted-pod-security-profile.md

```diff
@@ -0,0 +1,157 @@
+You can learn more about the Pod Security Admission in the [Kubernetes documentation](https://kubernetes.io/docs/concepts/security/pod-security-admission/)
+
+## Current situation
+
+Currently, Strimzi does by default only minimal configuration of the security context.
+When running on Kubernetes and using persistent storage, we configure the group which should be used for the storage.
+That is the only automatic configuration Strimzi does.
+
+Users can use the `.template` fields in the Strimzi custom resources to configure their own security context.
+That allows them to customize it to match their own requirements and policies.
+On OpenShift, the security context is automatically injected into the Pods when they are created based on the OpenShift SCC policies.
+This typically involves dropping some capabilities, using some random unprivileged user ID etc.
+In some cases, our users might also use similar systems to inject the security context using admission controllers and similar tools.
+
+The _baseline_ profile basically means that you just use the default settings without requesting any additional privileges or capabilities.
+So without any of the additional tooling mentioned above, Strimzi operators and the pods created by them today match the _baseline_ profile.
+But if you would try to run Strimzi in a cluster where the _restricted_ profile is enforced, it would not work out of the box.
+
+Apart from the Kaniko builder used for the Kafka Connect build, all our components are able to run under the _restricted_ profile.
+But the operator doesn't configure the appropriate security context and the admission plugin will reject them.
+Users have to _manually_ add the matching security context configuration into the `.template` sections of the custom resources and into the operator deployments to allow the pods to be created under the _restricted_ profile.
+To match the _restricted_ profile, users would need to add the following security context:
+
+```yaml
+          securityContext:
+            allowPrivilegeEscalation: false
+            capabilities:
+              # The restricted allows the use of the NET_BIND_SERVICE capability.
+              # But Strimzi does not require it for anything, so we do not include it here.
+              drop:
+                - ALL
+            runAsNonRoot: true
+            seccompProfile:
+              type: RuntimeDefault
+```
+
+While users can configure the security context, it is not user-friendly.
+The Pod security context does not cover all the different options required by the _restricted_ profile.
+So the security context needs to be basically configured for every container (including init containers, sidecars etc.).
+So just for a single Kafka cluster, it might need to be configured on 9 different places of the Kafka CR (ZooKeeper, Kafka broker, Kafka init-container, Topic Operator, User Operator, TLS sidecar, Kafka Exporter, Cruise Control, another TLS sidecar).
+
+## Proposal
```

Comment:
Do we want a list of what needs to change? _The following steps should be taken as part of the 0.28.0 release:_

Reply:
I would avoid going into the code-level details. They do not really affect the user value of this feature and I do not think the implementation it self contains anything complicated which would need to be covered here.